packages/edit-site/src/components/sidebar-edit-mode/template-card/last-revision.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { PanelBody, Button } from '@wordpress/components';
+import { sprintf, _n } from '@wordpress/i18n';
+import { backup } from '@wordpress/icons';
+import { addQueryArgs } from '@wordpress/url';
+import { PostTypeSupportCheck } from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../../store';
+
+const useRevisionData = () => {
+	const { currentTemplate } = useSelect( ( select ) => {
+		const { getCurrentTemplate } = select( editSiteStore );
+		return {
+			currentTemplate: getCurrentTemplate(),
+		};
+	} );
+
+	const lastRevisionId =
+		currentTemplate?._links?.[ 'predecessor-version' ]?.[ 0 ]?.id ?? null;
+
+	const revisionsCount =
+		( currentTemplate?._links?.[ 'version-history' ]?.[ 0 ]?.count ?? 0 ) +
+		1;
+
+	return {
+		lastRevisionId,
+		revisionsCount,
+	};
```

Comment:
Suggestion: These values can be fetched/assigned inside `mapSelect`; there's less chance of extra rerenders when comparing the return object with primitive types. What do you think?

Reply:
Good suggestion added in [7fdd6cc](https://github.com/WordPress/gutenberg/pull/45215/commits/7fdd6ccb541baff2b8138be795ea4cf6b592c66f)